core/manifest/ConfigStaticManifest.php

```diff
@@ -199,6 +199,25 @@ protected function next($ignoreWhitespace = true) {
 	}
 
 	/**
+	 * Get the previous token processed. Does *not* decrement the pointer
+	 *
+	 * @param bool $ignoreWhitespace - if true will skip any whitespace tokens & only return non-whitespace ones
+	 * @return null | mixed - Either the previous token or null if there isn't one
+	 */
+	protected function lastToken($ignoreWhitespace = true) {
+		// Subtract 1 as the pointer is always 1 place ahead of the current token
+		$pos = $this->pos - 1;
```

Comment:
This is because the pointer (`$this->pos`) doesn’t actually match up to the current token being inspected, see comment added below:

```php
protected function next($ignoreWhitespace = true) {
	do {
		if($this->pos >= $this->length) return null;
		// If current position is 1, this will fetch the 1st token, *then* increment the pointer
		$next = $this->tokens[$this->pos++];
	}
	while($ignoreWhitespace && is_array($next) && $next[0] == T_WHITESPACE);

	return $next;
}
```

So `next()` will return the n-th token, but set the pointer to be n+1. I didn’t fancy refactoring this, as changing it affects pretty much everything else in the parser, plus it’s being removed in 4.x anyway.